### ISSUE TITLE
add `deviceScaleFactor` support

### ISSUE THREAD

**Issue by AnnatarHe** [jasonraimondi/url-to-png#12]
puppeteer support `deviceScaleFactor` for high resolution screenshot. can we support that parameter?

I notice that maybe we can change this line: https://github.com/jasonraimondi/url-to-png/blob/master/src/services/image-render.service.ts#L62

**Comment by jasonraimondi** [comment]
@AnnatarHe I just added a branch called `device-scale-factor` that supports this field, but in my testing, I did not see any output difference of screenshots. Do you mind testing and seeing if this does what you are expecting?

**Comment by AnnatarHe** [comment]
I clone the repo and switch to branch `device-scale-factor`

and tried to download
```
http://localhost:3000/?url=https%3A%2F%2Fjasonraimondi.com&isMobile=true&isFullPage=true&viewPortWidth=375&width=375&deviceScaleFactor=1
```
by change `width` and `deviceScaleFactor` with `width=375`, `width=1125` and `deviceScaleFactor=1`, `deviceScaleFactor=3`

and Airdrop to iPhone 12 and try to scale image

In my eyes I think `width=1125` and `deviceScaleFactor=3` has best quality

* 1125 with 1 dpi
![1125-1factor](https://user-images.githubusercontent.com/8704175/110065892-365ee380-7dab-11eb-8ac1-4b7990bd7934.png)

* 1125 with 3 dpi
![1125-3factor](https://user-images.githubusercontent.com/8704175/110065920-48d91d00-7dab-11eb-9868-a7b2ba7bbb4f.png)

* 375 with 3 dpi
![3factor](https://user-images.githubusercontent.com/8704175/110065926-4aa2e080-7dab-11eb-8e72-109ff6e4a9a2.png)

* 375 with 1 dpi
![1factor](https://user-images.githubusercontent.com/8704175/110065927-4b3b7700-7dab-11eb-92f1-5d8f67190265.png)

**Comment by jasonraimondi** [comment]
Oh awesome thank you for the great update! I'll go ahead and merge this into master and get a tag out for v1.2.0

**Comment by jasonraimondi** [comment]
merged #13
